configure.ac

```diff
@@ -152,10 +152,18 @@ AS_IF([test x$enable_http2 != xno ], [
 dnl When bumping the libsoup-2.4 dependency, remember to bump
 dnl SOUP_VERSION_MIN_REQUIRED and SOUP_VERSION_MAX_ALLOWED in
 dnl Makefile.am
-SOUP_DEPENDENCY="libsoup-2.4 >= 2.39.1"
+SOUP_DEPENDENCY="libsoup-3.0 >= 3.0.0"
+SOUP2_DEPENDENCY="libsoup-2.4 >= 2.39.1"
 AC_ARG_WITH(soup,
 	    AS_HELP_STRING([--with-soup], [Use libsoup @<:@default=yes@:>@]),
 	    [], [with_soup=$with_soup_default])
+AC_ARG_WITH(soup2,
+	    AS_HELP_STRING([--with-soup2], [Use libsoup2 @<:@default=no@:>@]),
+	    [], [with_soup2=no])
+AS_IF([test x$with_soup2 != xno], [
+    SOUP_DEPENDENCY=$SOUP2_DEPENDENCY
+    with_soup=yes
+])
```

Comment:
I think `--with-soup` should continue to be soup2 and the default. The soup3 backend should be a new option `--with-soup3`. I don't think it makes sense to switch to a brand new backend for the default. The soup2 and curl backends have seen years of testing in the field and I don't think "because soup3 is available" is a good reason to switch that right away. The goal to eradicate soup2 from a system has merit, but having a tested and working fetch backend is likely a much stronger goal for most people. As I stated before, if you're using ostree for a bootable OS, you don't get updates if the fetch backend has bugs. That's the worst kind of bug.

When the soup3 backend gets some testing in the field, we can flip around the options, IMO.

Reply:
ok, that's fair, we can do that